test/kb_blast_server_test.py

```diff
@@ -71,33 +71,156 @@ def getImpl(self):
+                name = self.__class__.genomeName_list[item_i]
+                if info != None:
+                    if name != genome_basename:
+                        self.__class__.genomeInfo_list[item_i] = None
+                        self.__class__.genomeName_list[item_i] = None
+                    else:
+                        return info
+            except:
+                pass
+
+        # 1) transform genbank to kbase genome object and upload to ws
+        shared_dir = "/kb/module/work/tmp"
+        genome_data_file = 'data/genomes/'+genome_basename+'.gbff.gz'
+        genome_file = os.path.join(shared_dir, os.path.basename(genome_data_file))
+        shutil.copy(genome_data_file, genome_file)
+
+        SERVICE_VER = 'release'
+        #SERVICE_VER = 'dev'
+        GFU = GenomeFileUtil(os.environ['SDK_CALLBACK_URL'],
+                             token=self.getContext()['token'],
+                             service_ver=SERVICE_VER
+                         )
+        print ("UPLOADING genome: "+genome_basename+" to WORKSPACE "+self.getWsName()+" ...")
+        genome_upload_result = GFU.genbank_to_genome({'file': {'path': genome_file },
+                                                      'workspace_name': self.getWsName(),
+                                                      'genome_name': genome_basename
+                                                  })
+#                                                  })[0]
+        pprint(genome_upload_result)
+        genome_ref = genome_upload_result['genome_ref']
+        new_obj_info = self.getWsClient().get_object_info_new({'objects': [{'ref': genome_ref}]})[0]
+
+        # 2) store it
+        if not hasattr(self.__class__, 'genomeInfo_list'):
+            self.__class__.genomeInfo_list = []
+            self.__class__.genomeName_list = []
+        for i in range(item_i+1):
+            try:
+                assigned = self.__class__.genomeInfo_list[i]
+            except:
+                self.__class__.genomeInfo_list.append(None)
+                self.__class__.genomeName_list.append(None)
+
+        self.__class__.genomeInfo_list[item_i] = new_obj_info
+        self.__class__.genomeName_list[item_i] = genome_basename
+        return new_obj_info
+
+    # call this method to get the WS object info of an AnnotatedMetagenomeAssembly
+    #   (will upload the example data if this is the first time the method is called during tests)
+    def getAMAInfo(self, ama_basename, item_i=0):
+        if hasattr(self.__class__, 'amaInfo_list'):
+            try:
+                info = self.__class__.amaInfo_list[item_i]
+                name = self.__class__.amaName_list[item_i]
+                if info != None:
+                    if name != ama_basename:
+                        self.__class__.amaInfo_list[item_i] = None
+                        self.__class__.amaName_list[item_i] = None
+                    else:
+                        return info
```

Comment:
can you abstract this out and make it a general method, since it's used in a couple of places?

e.g.

```py
def getInfoForType(self, type, basename, item_i=0):
    infoAttr = type + 'Info_list' # e.g. 'ama' or 'genome'
    if hasattr(self.__class__, infoAttr):
            nameAttr = type + 'Name_list'
            try:
                 info = getattr(self.__class__, infoAttr)[item_i]
                 name = getattr(self.__class__, nameAttr)[item_i]
  # (etc.)
```

Reply:
done